components/browser/engine-system/src/main/java/mozilla/components/browser/engine/system/SystemEngineView.kt

```diff
@@ -694,15 +695,12 @@ class SystemEngineView @JvmOverloads constructor(
 
     override fun canScrollVerticallyDown() = session?.webView?.canScrollVertically(1) ?: false
 
-    @Suppress("Deprecation")
-    override fun getInputResult(): EngineView.InputResult {
-        (session?.webView as? NestedWebView)?.let { nestedWebView ->
-            // Direct mapping of WebView's returned values.
-            // There should be a 1-1 relation. If not fail fast to allow for a quick fix.
-            return EngineView.InputResult.values().first { it.value == nestedWebView.inputResult }
-        }
-        // Let's be preventive about not knowing if user's touch was handled and say no.
-        return EngineView.InputResult.INPUT_RESULT_UNHANDLED
+    override fun getInputResultDetail(): InputResultDetail {
+        return (session?.webView as? NestedWebView)?.inputResultDetail
+            ?: InputResultDetail().apply {
+                // The constructor enables vertical overscrolling by default. Don't want that.
+                update(overscrollDirections = 0)
```

Comment:
nit: Should we use a constant instead of a direct value?

Reply:
Would've preferred the InputResultDetail class to not expose these constants since they should be of no use to clients.
GV sends the right data (because this api if modelled after GV) but WV doesn't.
I agree hardcoding this value here doesn't look nice. Will change.

Reply:
Maybe we can add a comment to clarify what `0` means here.

Reply:
After refactoring `InputResultDetail` this call is now just
>         return (session?.webView as? NestedWebView)?.inputResultDetail
>            ?: InputResultDetail.newInstance()
so I think the initial issue is now cleared.

Reply:
👍🏽